server/src/main/java/com/objectcomputing/checkins/services/action_item/ActionItemServicesImpl.java

```diff
@@ -1,79 +1,131 @@
 import java.util.Set;
 import java.util.UUID;
 
 import static com.objectcomputing.checkins.util.Util.nullSafeUUIDToString;
 
+@Singleton
 public class ActionItemServicesImpl implements ActionItemServices {
 
-    @Inject
-    private CheckInRepository checkinRepo;
-    @Inject
-    private ActionItemRepository actionItemRepo;
-    @Inject
-    private MemberProfileRepository memberRepo;
+    private final CheckInServices checkInServices;
+    private final ActionItemRepository actionItemRepo;
+    private final MemberProfileServices memberServices;
+    private final SecurityService securityService;
+    private final CurrentUserServices currentUserServices;
+    private final Validation validation;
+
+    public ActionItemServicesImpl(CheckInServices checkInServices, ActionItemRepository actionItemRepo,
+                                  MemberProfileServices memberServices, SecurityService securityService,
+                                  CurrentUserServices currentUserServices, Validation validation) {
+        this.checkInServices = checkInServices;
+        this.actionItemRepo = actionItemRepo;
+        this.memberServices = memberServices;
+        this.securityService = securityService;
+        this.currentUserServices = currentUserServices;
+        this.validation = validation;
+    }
 
     public ActionItem save(ActionItem actionItem) {
         ActionItem actionItemRet = null;
+        String workEmail = securityService != null ? securityService.getAuthentication().get().getAttributes().get("email").toString() : null;
+        MemberProfile currentUser = workEmail != null ? currentUserServices.findOrSaveUser(null, workEmail) : null;
+        Boolean isAdmin = securityService != null && securityService.hasRole(RoleType.Constants.ADMIN_ROLE);
+
         if (actionItem != null) {
-            final UUID guildId = actionItem.getCheckinid();
-            final UUID createById = actionItem.getCreatedbyid();
+            final UUID checkinId = actionItem.getCheckinid();
+            final UUID createdById = actionItem.getCreatedbyid();
+
+            CheckIn checkinRecord = checkInServices.read(checkinId);
+            Boolean isCompleted = checkinRecord != null ? checkinRecord.isCompleted() : null;
+            final UUID pdlId = checkinRecord != null ? checkinRecord.getPdlId() : null;
+            final UUID teamMemberId = checkinRecord != null ? checkinRecord.getTeamMemberId() : null;
+
+            validation.validateArguments(checkinId == null || createdById == null, "Invalid action item %s", actionItem);
+            validation.validateArguments(actionItem.getId() != null, "Found unexpected id %s for action item", actionItem.getId());
+            validation.validateArguments(checkInServices.read(checkinId) == null, "CheckIn %s doesn't exist", checkinId);
+            validation.validateArguments(memberServices.getById(createdById) == null, "Member %s doesn't exist", createdById);
+
+            if (!isAdmin && isCompleted) {
```

Comment:
Let's use `validatePermissions` to do this validation, rather than doing the validation here and just having that method throw an exception. For example a `validateActionItemsPermissions` method that does all this.